language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java

```diff
@@ -484,7 +487,10 @@ protected LSCompletionItem getImplicitNewCItemForClass(ClassSymbol symbol, Balle
     }
 
     protected List<LSCompletionItem> expressionCompletions(BallerinaCompletionContext context) {
-        List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition());
```

Comment:
We should be updating the `DocumentServiceContext.visibleSymbols()` method instead of using the `semanticModel` directly. In `DocumentServiceContext`, we should overload, `visibleSymbols()` method to take in `DiagnosticState` (vararg) as a second parameter and update the existing `visibleSymbols()` method to pass in `VALID` as the diagnostic state.

Reply:
Should we add the same logic to WorkSpaceServiceContext as well? I'm not sure about it. @IMS94 WDYT?

Reply:
Yes. +1. It as only one usage (in CreateFunctionExecutor) right now. But, we should do the same there as well.